src/DependencyInjection/AlgoliaSearchExtension.php

```diff
@@ -71,7 +73,11 @@ public function load(array $configs, ContainerBuilder $container)
                 $engineDefinition,
                 $config,
             ]
-        ))->setPublic(true);
+        ));
+
+        $searchServiceDefinitionForAtomicReindex = (clone $searchServiceDefinition)
+            ->replaceArgument(2, ['prefix' => 'atomic_reindex_in_progress_' . $config['prefix']] + $config)
```

Comment:
Can we make this `atomic_reindex_in_progress_` a random name? Example:
```php
'atomic_temporary_'.uniqid('php_', true)
```

Reply:
I don't think we can do it with your current test setup, as it seems you are whitelisting index names

Reply:
No worries, make a temporary index name, @chloelbn will make sure it works on the test suite.